File: Exercism/python/allergies/allergies.py
class Allergies(object):
    ALLERGENS = [
        "eggs",
        "peanuts",
        "shellfish",
        "strawberries",
        "tomatoes",
        "chocolate",
        "pollen",
        "cats"
    ]

    def __init__(self, score):
        self._score = score & 0xff

    def _check(self, index):
        return self._score & 1 << index

    def allergic_to(self, item):
        return bool(self._check(self.ALLERGENS.index(item)))

    @property
    def lst(self):
        return [self.ALLERGENS[i] for i in range(8) if self._check(i)]

File: Exercism/c/allergies/src/allergies.c
#include "allergies.h"
#include <string.h>


static inline unsigned int check(unsigned int score, unsigned int index) {
  return score & 1 << index;
}

bool is_allergic_to(allergen_t allergen, unsigned int score) {
  return check(score, allergen) && true;
}

allergen_list_t get_allergens(unsigned int score) {
  allergen_list_t allergens;
  memset(&allergens, 0, sizeof(allergen_list_t));
  for (unsigned int i = 0; i < ALLERGEN_COUNT; i++) {
    if (check(score, i)) {
      allergens.count++;
      allergens.allergens[i] = true;
    }
  }
  return allergens;
}
